updater.c

```diff
@@ -872,7 +872,7 @@ static int trail_get_new_steps(struct pantavisor *pv)
 	if (!errno && (new_rev <= current_rev)) {
 		pv_log(WARN, "running revision %ld but trying to update to %ld",
 		       current_rev, new_rev);
-		if (new_rev < current_rev) {
+		if (new_rev <= current_rev) {
```

Comment:
I think this should remain <, as we now load whatever is in /storage/trails/current/.pv/progress (updater.c:886). Then, if we find a final state (updater.c:903), we try to report that back, which is probably better than a generic "stale" error.

Reply:
This should not be the case since https://github.com/pantavisor/pantavisor/pull/481. The idea now is to report to hub whatever is on disk if we find something final (DONE, UPDATED, WONTGO or ERROR).